cmd/extensions/oidc-policy/internal/controller/oidcpolicy_reconciler.go

```diff
@@ -0,0 +1,616 @@
+	"github.com/kuadrant/kuadrant-operator/pkg/extension/utils"
+
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type OIDCPolicyReconciler struct {
+	*reconcilers.BaseReconciler
+	logger logr.Logger
+}
+
+type ingressGatewayInfo struct {
+	Hostname  string                    `json:"hostname"`
+	Name      string                    `json:"name"`
+	Namespace string                    `json:"namespace"`
+	Protocol  gatewayapiv1.ProtocolType `json:"protocol"`
+	url       *url.URL
+}
+
+func (g *ingressGatewayInfo) GetURL() *url.URL {
+	if g.url == nil {
+		g.url = &url.URL{
+			Scheme: strings.ToLower(string(g.Protocol)),
+			Host:   g.Hostname,
+		}
+	}
+	return g.url
+}
+
+func NewOIDCPolicyReconciler() *OIDCPolicyReconciler {
+	return &OIDCPolicyReconciler{
+		BaseReconciler: reconcilers.NewLazyBaseReconciler(),
+	}
+}
+
+func (r *OIDCPolicyReconciler) WithLogger(logger logr.Logger) *OIDCPolicyReconciler {
+	r.logger = logger
+	return r
+}
+
+// kuadrant permissions
+//+kubebuilder:rbac:groups=kuadrant.io,resources=oidcpolicies,verbs=get;list;watch;update;patch
+//+kubebuilder:rbac:groups=kuadrant.io,resources=oidcpolicies/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kuadrant.io,resources=oidcpolicies/finalizers,verbs=update
+//+kubebuilder:rbac:groups=kuadrant.io,resources=authpolicies,verbs=get;create;list;watch;update;patch
+//+kubebuilder:rbac:groups=kuadrant.io,resources=authpolicies/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=kuadrant.io,resources=authpolicies/finalizers,verbs=update
+
+// gateway-api permissions
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=httproutes,verbs=get;create;list;watch;update;patch
+//+kubebuilder:rbac:groups=gateway.networking.k8s.io,resources=httproutes/status,verbs=get;update;patch
+
+// TODO: Current OIDC Workflow works only for Browser apps, needs implementation for Native apps
+
+func (r *OIDCPolicyReconciler) Reconcile(ctx context.Context, request reconcile.Request, kCtx types.KuadrantCtx) (reconcile.Result, error) {
+	r.WithLogger(utils.LoggerFromContext(ctx).WithName("OIDCPolicyReconciler"))
+	r.logger.Info("Reconciling OIDCPolicy")
+
+	oidcPolicy := &kuadrantv1alpha1.OIDCPolicy{}
+	if err := r.Client().Get(ctx, request.NamespacedName, oidcPolicy); err != nil {
+		if errors.IsNotFound(err) {
+			r.logger.Error(err, "OIDCPolicy not found")
+			return reconcile.Result{}, nil
+		}
+		r.logger.Error(err, "Failed to get OIDCPolicy")
+		return reconcile.Result{}, err
+	}
+
+	if oidcPolicy.GetDeletionTimestamp() != nil {
+		r.logger.Info("OIDCPolicy marked to be deleted")
+		return ctrl.Result{}, nil
+	}
+
+	ingressGatewayData, err := extcontroller.Resolve[ingressGatewayInfo](ctx, kCtx, oidcPolicy, "{\"protocol\": self.findGateways()[0].spec.listeners[0].protocol, \"hostname\": self.findGateways()[0].spec.listeners[0].hostname, \"name\": self.findGateways()[0].metadata.name, \"namespace\": self.findGateways()[0].metadata.namespace}", true)
```

Comment:
Not as bad as I feared... obviously, this could fail if no `Gateway` found still, which I guess is handled by line 111... but 👌 
Maybe be have the "main" string use "`` ` ``" delimiters? or inner ones use "`'`"?... just to make this cleaner... possibly with new lines? Just as this is to be used to illustrate, this part could be "clearer" to a reader.